lib/apiservers/service/restapi/handlers/vch_get.go

```diff
@@ -0,0 +1,428 @@
+			Limit:       asMiB(d.ResourceLimits.VCHMemoryLimitsMB),
+			Reservation: asMiB(d.ResourceLimits.VCHMemoryReservationsMB),
+			Shares:      asShares(d.ResourceLimits.VCHMemoryShares),
+		},
+		Resource: &models.ManagedObject{
+			ID: mobidToId(vchConfig.Container.ComputeResources[0].String()),
+		},
+	}
+
+	// network
+	model.Network = &models.VCHNetwork{
+		Bridge: &models.VCHNetworkBridge{
+			PortGroup: &models.ManagedObject{
+				ID: mobidToId(vchConfig.ExecutorConfig.Networks[vchConfig.Network.BridgeNetwork].Network.Common.ID),
+			},
+			IPRange: asIPRange(vchConfig.Network.BridgeIPRange),
+		},
+		Client:     asNetwork(vchConfig.ExecutorConfig.Networks["client"]),
+		Management: asNetwork(vchConfig.ExecutorConfig.Networks["management"]),
+		Public:     asNetwork(vchConfig.ExecutorConfig.Networks["public"]),
+	}
+
+	containerNetworks := make([]*models.ContainerNetwork, 0, len(vchConfig.Network.ContainerNetworks))
+	for key, value := range vchConfig.Network.ContainerNetworks {
+		if key != "bridge" {
+			containerNetworks = append(containerNetworks, &models.ContainerNetwork{
+				Alias: value.Name,
+				PortGroup: &models.ManagedObject{
+					ID: mobidToId(value.Common.ID),
+				},
+				Nameservers: *asIPAddresses(&value.Nameservers),
+				Gateway: &models.Gateway{
+					Address:             asIPAddress(value.Gateway.IP),
+					RoutingDestinations: []models.IPRange{asIPRange(&value.Gateway)},
+				},
+				IPRanges: *asIPRanges(&value.Destinations),
+			})
+		}
+	}
+	model.Network.Container = containerNetworks
+
+	// storage
+	scratchSize := int(vchConfig.Storage.ScratchSize)
+	model.Storage = &models.VCHStorage{
+		BaseImageSize: asKiB(&scratchSize),
+	}
+
+	volumeLocations := make([]*models.VCHStorageVolumeStoresItems0, 0, len(vchConfig.Storage.VolumeLocations))
+	for label, path := range vchConfig.Storage.VolumeLocations {
+		volume := models.VCHStorageVolumeStoresItems0{Datastore: path.String(), Label: label}
+		volumeLocations = append(volumeLocations, &volume)
+	}
+	model.Storage.VolumeStores = volumeLocations
+
+	imageStores := make([]string, 0, len(vchConfig.Storage.ImageStores))
+	for _, value := range vchConfig.Storage.ImageStores {
+		imageStores = append(imageStores, value.String())
+	}
+	model.Storage.ImageStores = imageStores
+
+	// auth
+	model.Auth = &models.VCHAuth{
+		Client: &models.VCHAuthClient{},
+	}
+
+	if vchConfig.Certificate.HostCertificate != nil {
+		model.Auth.Server = &models.VCHAuthServer{
+			Certificate: asPemCertificate(vchConfig.Certificate.HostCertificate.Cert),
+		}
+	}
+
+	model.Auth.Client.CertificateAuthorities = asPemCertificates(vchConfig.Certificate.CertificateAuthorities)
+
+	// endpoint
+	model.Endpoint = &models.VCHEndpoint{
+		UseResourcePool: d.UseRP,
+		Memory:          asMiB(&d.MemoryMB),
+		CPU: &models.VCHEndpointCPU{
+			Sockets: int64(d.NumCPUs),
+		},
+		OperationsCredentials: &models.VCHEndpointOperationsCredentials{
+			User: vchConfig.Connection.Username,
```

Comment:
Worth excluding username as well?

Reply:
I think including the operations usename is useful, and potentially important.

If the VCH suddenly stops working, being able to look at it and say "Oh, it's using my account, and I just changed the password!" or "Oh, it was using Joe's account, but... we just fired Joe and deactivated his account." might be helpful. (Or, alternatively, to verify that it's using a service account, as expected, and _not_ Joe's account.)